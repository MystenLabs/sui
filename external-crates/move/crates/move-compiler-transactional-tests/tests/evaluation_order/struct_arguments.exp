processed 6 tasks

task 1 'run'. lines 40-47:
Error: Function execution failed with VMError: {
    major_status: ARITHMETIC_ERROR,
    sub_status: None,
    location: 0x42::M,
    indices: [],
    offsets: [(FunctionDefinitionIndex(0), 2)],
}

task 2 'run'. lines 49-56:
Error: Function execution failed with VMError: {
    major_status: ARITHMETIC_ERROR,
    sub_status: None,
    location: 0x42::M,
    indices: [],
    offsets: [(FunctionDefinitionIndex(1), 2)],
}

task 3 'run'. lines 58-65:
Error: Function execution failed with VMError: {
    major_status: ARITHMETIC_ERROR,
    sub_status: None,
    location: 0x42::M,
    indices: [],
    offsets: [(FunctionDefinitionIndex(2), 2)],
}

task 4 'run'. lines 67-74:
Error: Function execution failed with VMError: {
    major_status: ARITHMETIC_ERROR,
    sub_status: None,
    location: 0x42::M,
    indices: [],
    offsets: [(FunctionDefinitionIndex(3), 2)],
}

task 5 'run'. lines 76-83:
Error: Function execution failed with VMError: {
    major_status: ARITHMETIC_ERROR,
    sub_status: None,
    location: 0x42::M,
    indices: [],
    offsets: [(FunctionDefinitionIndex(4), 2)],
}
