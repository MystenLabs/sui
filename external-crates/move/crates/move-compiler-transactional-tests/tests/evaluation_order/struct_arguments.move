//# publish
module 0x42::M {
    struct S has drop {
        a: u64,
        b: u64,
    }

    struct R has key, store {}
    struct Cup has key {
        a: u64,
        b: R,
    }

    public fun t0() {
        S { b: 1 / 0, a: fail(0) };
    }

    public fun t1() {
        S { b: 18446744073709551615 + 18446744073709551615, a: fail(0) };
    }

    public fun t2() {
        S { b: 0 - 1, a: fail(0) };
    }

    public fun t3() {
        S { b: 1 % 0, a: fail(0) };
    }

    public fun t4() {
        S { b: 18446744073709551615 * 18446744073709551615, a: fail(0) };
    }

    fun fail(code: u64): u64 {
        abort code
    }

}

//# run
module 1::m {
use 0x42::M;
fun main() {
  // arithmetic error
  M::t0()
}
}

//# run
module 2::m {
use 0x42::M;
fun main() {
  // arithmetic error
  M::t1()
}
}

//# run
module 3::m {
use 0x42::M;
fun main() {
  // arithmetic error
  M::t2()
}
}

//# run
module 4::m {
use 0x42::M;
fun main() {
  // arithmetic error
  M::t3()
}
}

//# run
module 5::m {
use 0x42::M;
fun main() {
  // arithmetic error
  M::t4()
}
}
