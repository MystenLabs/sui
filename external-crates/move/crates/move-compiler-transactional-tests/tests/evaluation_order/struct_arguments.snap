processed 6 tasks

task 1, lines 40-47:
//# run
Error: Function execution failed with VMError: {
    major_status: ARITHMETIC_ERROR,
    sub_status: None,
    location: 0x42::M,
    indices: [],
    offsets: [(FunctionDefinitionIndex(0), 2)],
}

task 2, lines 49-56:
//# run
Error: Function execution failed with VMError: {
    major_status: ARITHMETIC_ERROR,
    sub_status: None,
    location: 0x42::M,
    indices: [],
    offsets: [(FunctionDefinitionIndex(1), 2)],
}

task 3, lines 58-65:
//# run
Error: Function execution failed with VMError: {
    major_status: ARITHMETIC_ERROR,
    sub_status: None,
    location: 0x42::M,
    indices: [],
    offsets: [(FunctionDefinitionIndex(2), 2)],
}

task 4, lines 67-74:
//# run
Error: Function execution failed with VMError: {
    major_status: ARITHMETIC_ERROR,
    sub_status: None,
    location: 0x42::M,
    indices: [],
    offsets: [(FunctionDefinitionIndex(3), 2)],
}

task 5, lines 76-83:
//# run
Error: Function execution failed with VMError: {
    major_status: ARITHMETIC_ERROR,
    sub_status: None,
    location: 0x42::M,
    indices: [],
    offsets: [(FunctionDefinitionIndex(4), 2)],
}
